src/cpp/lpnamer/model/ActiveLinks.cpp

```diff
@@ -162,11 +186,25 @@ const std::vector<Candidate>& ActiveLink::getCandidates() const {
 }
 
 double ActiveLink::already_installed_direct_profile(size_t timeStep) const {
-  return _already_installed_profile.getDirectProfile(timeStep);
+  return _already_installed_profile.at(0).getDirectProfile(timeStep);
 }
 
 double ActiveLink::already_installed_indirect_profile(size_t timeStep) const {
-  return _already_installed_profile.getIndirectProfile(timeStep);
+  return _already_installed_profile.at(0).getIndirectProfile(timeStep);
+}
+
+double ActiveLink::already_installed_direct_profile(size_t chronicle_number, size_t timeStep) const {
+  if (chronicle_number == 0) chronicle_number = 1;
+  if (chronicle_number > _already_installed_profile.size())
+    chronicle_number = 1;
+  return _already_installed_profile.at(chronicle_number - 1).getDirectProfile(timeStep);
+}
+
+double ActiveLink::already_installed_indirect_profile(size_t chronicle_number, size_t timeStep) const {
+  if (chronicle_number == 0) chronicle_number = 1;
+  if (chronicle_number > _already_installed_profile.size())
+    chronicle_number = 1;
+  return _already_installed_profile.at(chronicle_number - 1).getIndirectProfile(timeStep);
 }
```

Comment:
To avoid code duplication, I suggest using a single function for both directions, passing enum-class LinkDirection (direct/indirect)

Reply:
It is a good suggestion. I'd prefer not to implement it for now because the whole profile management system (or even lp namer) need a good rethinking. We often work on both values (direct and indirect) at a time so it would be good to think of an interface that make it seamless for clients if it is possible.